include/crpropa/PhotonBackground.h

```diff
@@ -65,7 +65,7 @@ class PhotonField: public Referenced {
  */
 class TabularPhotonField: public PhotonField {
 public:
-	TabularPhotonField(const std::string fieldName, const bool isRedshiftDependent);
+	TabularPhotonField(const std::string fieldName, const bool isRedshiftDependent = true);
```

Comment:
I am not sure if this is connected to ur discussion of PR #255?!
We decided there to have the non-tabulated redshift evolution as the default because of possible inconsistencies between modules.

Is that inconsitency resolved by this PR or is there any other reason to change the default behavior or am I missing something?

Reply:
well it can be but here this is more involved:

1) ```isRedshiftDependent``` is a property of ```(Tabular)PhotonField``` and helps that class to call appropriate interpolation routines, check input data, calc redshift scalings, etc. which all are properties of the new photon field structures, too
2) Note how all built-in photon fields (aka. all the IRB models) have that property already by being instantiated with ```isRedshiftDependent = true```.
3) #255 in return refers to a separate property ```haveRedshiftDependence``` in ```PhotoPionProduction``` which the user actually is free to choose. The main reason here is, I guess, that the user may choose between the 1D and the more expensive 2D interpolation for runtime reasons (?).
4) In my eyes, that parameter should be abolished and redshift dependence could now be inferred from the photon field at hand. I did not crunch the numbers here but the most severe concern of runtime in the PPP is the threading bottleneck of SOPHIA and less the choice of 1D or 2D interpolations.

Reply:
I mixed up `isRedshiftDepentend` with `haveRedshiftDependence`.
So basically my issue concerning this PR is solved.

In addition:
4) Abolishing that and taking that value from the Photonfield class is in general a good idea. 

My original concern here was, that (at least in the current master) we do not have 2 dimensional tabulated files for the rates for other modules than PPP. So the reason for choosing 1d over 2d might not only be runtime driven but also by wishing for consistency between different interaction losses, e.g. PPP (2 dimensional rates available) and Photodisintegration (only 1 dimensional rates available).